Show status for proposals in table

Devorein opened the pull request.

mattcasey commented on components/proposals/components/ProposalsTable.tsx:
```diff
@@ -0,0 +1,119 @@
+import { ProposalWithUsers } from 'lib/proposal/interface';
+import { humanFriendlyDate, toMonthDate } from 'lib/utilities/dates';
+import { useCallback, useState } from 'react';
+import { ProposalStatusChip } from '../../../components/[pageId]/DocumentPage/components/ProposalProperties/ProposalStatusBadge';
```
It would be better to put shared components inside the components/proposals folder (so move ProposalStatusBadge to here)

mattcasey commented on components/proposals/components/ProposalsTable.tsx:
```diff
@@ -0,0 +1,119 @@
+        ) : null;
+      })}
+      {activePage && (
+        <PageDialog
```
replace with `usePageDialog`

Devorein replied:
Wow. Its so much easier now 🤯 Updated 🥳

mattcasey commented on lib/proposal/getProposalsBySpace.ts:
```diff
@@ -0,0 +1,37 @@
+        }]
+      }
+    },
+    include: {
```
this should be joined with contributors on the frontend

Devorein replied:
It is joined on the frontend with the contributors based on the userId and roleId, we are not getting any other user data (apart from the id) when joining on the backend

mattcasey commented on pages/[domain]/proposals/index.tsx:
```diff
@@ -0,0 +1,15 @@
+import getPageLayout from 'components/common/PageLayout/getLayout';
+import { setTitle } from 'hooks/usePageTitle';
+
+export default function VotesPage () {
```
fix "VotesPage"

Devorein replied:
Oops. Thank you for that. Updated.

mattcasey commented on components/proposals/ProposalsPage.tsx:
```diff
@@ -0,0 +1,50 @@
+  const [currentSpace] = useCurrentSpace();
+
+  const { data, mutate: mutateProposals } = useSWR(() => `proposals/${currentSpace?.id}`, () => currentSpace ? charmClient.proposals.getProposalsBySpace(currentSpace.id) : [], {
+    fallbackData: undefined
```
i don't think this does anything since you're passing in undefined?

Devorein replied:
I copied most of the code from the `VotesPage` component, it was also used there. Removed it from both places as swr by default returns `undefined`

mattcasey commented on components/proposals/ProposalsPage.tsx:
```diff
@@ -0,0 +1,50 @@
+export default function ProposalsPage () {
+  const [currentSpace] = useCurrentSpace();
+
+  const { data, mutate: mutateProposals } = useSWR(() => `proposals/${currentSpace?.id}`, () => currentSpace ? charmClient.proposals.getProposalsBySpace(currentSpace.id) : [], {
```
it's better to return `null` in the first input to useSWR, until currentSpace exists, so that the 2nd function doesn't even get called

Devorein replied:
Updated both in `VotesPage` and `ProposalsPage`

mattcasey reviewed (changes requested):
Before this PR is merged, please remove the aggregated user data from ProposalWithUsers - u may want to get rid of this interface completely. we should just be using the hook 'useContributors' to retrieve user data

Devorein commented on components/proposals/components/ProposalsTable.tsx:
```diff
@@ -0,0 +1,119 @@
+  async function deleteProposal (proposalId: string) {
+    const page = pages[proposalId];
+    if (page) {
+      await charmClient.archivePage(page.id);
```
Updated.

Devorein commented on components/proposals/ProposalsPage.tsx:
```diff
@@ -0,0 +1,50 @@
+    fallbackData: undefined
+  });
+
+  // For now, consider that empty pages list means we are loading pages
```
The comment was actually there before this PR 👀

Devorein commented:
Right now we are just fetching the `userId` (which is a must) and the `proposalId` for authors and additional `roleId` for reviewers. No data about the author and reviewer is being fetched other than that and we join the data on the frontend using the id.

Devorein commented:
I've added back the filter and sort options for the proposals table. Completely forgot I removed it when copying code from the Votes table. For the sort only Created at is added as an option (we could add more new options later so kept it for now) and as for the filter options its all the available status of the proposal along with 'all' to include proposals of any status
